metaserver/client.go

```diff
@@ -21,6 +29,49 @@ type Client struct {
 	client *http.Client
 }
 
+func (client *Client) GetAuthToken(privateKey *rsa.PrivateKey, authType string) (string, error) {
+	challengeURL := fmt.Sprintf("http://%[1]s/auth/%[2]s?%[2]sID=1", client.addr, authType)
+
+	// Get a challenge token
+	resp, err := client.client.Get(challengeURL)
+	if err != nil {
+		return "", err
+	}
+	var respMsg authorization.GetAuthChallengeResp
+	_ = json.NewDecoder(resp.Body).Decode(&respMsg)
+	token := respMsg.Nonce
+
+	// Sign the token
+	hashed := sha256.Sum256([]byte(token))
+
+	signature, err := rsa.SignPKCS1v15(rand.Reader, privateKey, crypto.SHA256, hashed[:])
+	if err != nil {
+		return "", err
+	}
+
+	// Encode the token and send it back to the server.
+	encoded := base64.URLEncoding.EncodeToString(signature)
+
+	respondURL := fmt.Sprintf("http://%[1]s/auth/%[2]s", client.addr, authType)
+	responseField := fmt.Sprintf("%sID", authType)
+	resp, err = client.client.PostForm(respondURL, url.Values{responseField: {"1"}, "signedNonce": {encoded}})
+	if err != nil {
+		return "", err
+	} else {
+		println(resp.StatusCode)
+		var b []byte
+		defer resp.Body.Close()
+		b, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			panic(err)
```

Comment:
Stay calm :)

Reply:
I'll update that to return the error. I was a little fast and loose with my error handling code while I was prototyping. :P